content/blog/overview-package-management-python.md

```diff
@@ -0,0 +1,363 @@
+if there is an exact match it should show it to you.
+The help text says I can enter the exact package name but typing `django` tries to autocomplete with the package names
+from the list and results in an error:
+
+```bash
+ > django
+Value "django" is invalid
+
+Enter package # to add, or the complete package name if it is not listed:
+ [ 0] django-bagou
+ [ 1] django-maro
+ ... 99 packages
+```
+
+Poetry will also create virtualenv automatically when running `install`, `add` or `remove` with the current version of `python`:
+you cannot currently create a virtualenv with a different version.
+
+You are supposed to activate a shell with `poetry shell` but it breaks if you using `$WORKON_HOME` for other tools:
+https://github.com/sdispater/poetry/issues/214 so I ended up ignoring the virtualenv related commands. I didn't spend time looking
+into it so there might be an easy fix but it should work out of the box with the rest of the ecosystem.
+
+If you try to install a package using `poetry add django`, you will notice that a `pyproject.lock` has been created
+and that some dependencies were added to `pyproject.toml`:
+
+```toml
+[tool.poetry.dependencies]
+python = "3.7"
+django = "^2.1"
+
+[tool.poetry.dev-dependencies]
+pytest = "^3.0"
+```
+
+As opposed to the other tools, `poetry` uses the semantic versioning to refer to dependencies versions. If you have used
+[Cargo](https://github.com/rust-lang/cargo) in Rust, you will feel at home. The issue is that many Python libraries do not use
+semantic versioning so it does feel a bit weird to use in this context.
+
+While trying to clean its cache to test the speed, I ran into a small UX issue.
+Poetry offers a command to clean the cache (`poetry cache:clear`) but it takes a path/string to a cache as an argument. A new
+user like me will not know what argument to give, especially since the command is not [in the documentation](https://poetry.eustace.io/docs/cli/).
+Looking into issues, we can see that we can run `poetry cache:clear --all pypi` but the documentation doesn't even mention `cache:clear` so it's not
+obvious.
+
+In terms of speed, a `poetry install` with the same list of packages as the others was done in 40s.
+Subsequent runs are below a second. The UI is very clean and clear about what was installed and their versions.
+
+Poetry also has `poetry update` to update dependencies and once again makes it clear what has changed.
+
+While this post is focused on backend applications, it is worth mentioning that Poetry apparently handles building and publishing
+libraries as well which is a big plus if you are a maintainer: one tool for all purposes!
+
+## My opinion
+
+I have only mentioned 3 tools but there are way more! I wouldn't be surprised if Python had the record of the highest
+number of tools dedicated to package and environment management for a programming language.
```

Comment:
Instead of
>  I wouldn't be surprised if Python had the record of the highest
number of tools dedicated to package and environment management for a programming language.

I would have said

>  I wouldn't be surprised if Python had the highest
number of tools dedicated to package and environment management of any programming language.